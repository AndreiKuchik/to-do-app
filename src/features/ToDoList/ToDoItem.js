import React from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { Draggable } from 'react-beautiful-dnd';
import { deleteToDo, completeToDo } from '../toDoListSlice';

import styles from './ToDoItem.module.css';

const ToDoItem = (props) => {
  const dispatch = useDispatch();

  return (
    <Draggable key={props.id} draggableId={props.id} index={props.index}>
      {(provided) => (
        <li
          ref={provided.innerRef}
          {...provided.draggableProps}
          {...provided.dragHandleProps}
        >
          <input
            type="checkbox"
            checked={props.isCompleted}
            onChange={() => dispatch(completeToDo(props.id))}
            className={styles.isCompletedCheckBox}
          />
          <div
            className={`${styles.toDoText} ${
              props.isCompleted && styles.crossed
            }`}
          >
            {props.text}
          </div>
          <button
            type="button"
            className={styles.deleteButton}
            onClick={() => dispatch(deleteToDo(props.id))}
          >
            Delete
          </button>
        </li>
      )}
    </Draggable>
  );
};

ToDoItem.propTypes = {
  id: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  isCompleted: PropTypes.bool.isRequired,
  index: PropTypes.number.isRequired,
};

export default ToDoItem;
