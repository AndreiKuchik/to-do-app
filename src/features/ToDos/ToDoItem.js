import React from 'react';
import PropTypes from 'prop-types';
import { deleteToDo, completeToDo } from '../toDoListSlice';
import { useAppDispatch } from '../../core/hooks';

import styles from './ToDoItem.module.css';

const ToDoItem = (props) => {
  const dispatch = useAppDispatch();

  return (
    <React.Fragment>
      <input
        role="todo-item-checkbox"
        type="checkbox"
        checked={props.isCompleted}
        onChange={() => dispatch(completeToDo(props.id))}
        className={styles.isCompletedCheckBox}
      />
      <div
        role="todo-item-text"
        className={`${styles.toDoText} ${props.isCompleted && styles.crossed}`}
      >
        {props.text}
      </div>
      <button
        type="button"
        role="todo-item-delete-button"
        className={styles.deleteButton}
        onClick={() => dispatch(deleteToDo(props.id))}
      >
        Delete
      </button>
    </React.Fragment>
  );
};

ToDoItem.propTypes = {
  id: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  isCompleted: PropTypes.bool.isRequired,
};

export default ToDoItem;
