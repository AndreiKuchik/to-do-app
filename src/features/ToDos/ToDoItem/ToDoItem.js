import React from 'react';
import PropTypes from 'prop-types';
import { deleteToDo, completeToDo } from '../../toDosActions';
import { useAppDispatch } from '../../../app/hooks';

import styles from './ToDoItem.module.css';
import Checkbox from '../../../components/Checkbox';

const ToDoItem = ({ id, text, isCompleted }) => {
  const dispatch = useAppDispatch();
  const checkboxHandler = () => dispatch(completeToDo(id));

  return (
    <React.Fragment>
      <Checkbox
        role="todo-item-checkbox"
        isChecked={isCompleted}
        onCheckBoxChange={checkboxHandler}
        styles={styles.toDoCheckbox}
      />
      <div
        role="todo-item-text"
        className={`${styles.toDoText} ${isCompleted && styles.crossed}`}
      >
        {text}
      </div>
      <button
        type="button"
        role="todo-item-delete-button"
        className={styles.deleteButton}
        onClick={() => dispatch(deleteToDo(id))}
      >
        Delete
      </button>
    </React.Fragment>
  );
};

ToDoItem.propTypes = {
  id: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  isCompleted: PropTypes.bool.isRequired,
};

export default ToDoItem;
