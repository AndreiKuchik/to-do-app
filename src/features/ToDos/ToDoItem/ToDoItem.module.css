.toDoText {
  display: inline-block;
  margin-left: 1rem;
  color: #a3a5b2;
}

.crossed {
  text-decoration: line-through;
  color: #36d7b7;
}

.toDoCheckbox {
  width: 1.3em;
  height: 1.3em;
  border-radius: 60%;
  background-color: white;
  vertical-align: middle;
  border: 1px solid #ddd;
  appearance: none;
  cursor: pointer;
}

.toDoCheckbox:checked {
  background-color: #36d7b7;
}

.deleteButton {
  margin-left: auto;
  cursor: pointer;
  padding: 0.5rem;
  border: 1px #36d7b7;
  background-color: #36d7b7;
  color: white;
  border-radius: 12px;
  margin-right: 1rem;
}
