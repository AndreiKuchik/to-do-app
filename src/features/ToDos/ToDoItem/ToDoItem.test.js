import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import ToDoItem from './ToDoItem';

const mockStore = configureStore([]);
const store = mockStore({});

const props = {
  id: '1',
  index: 0,
  isCompleted: false,
  text: 'ToDo1',
};

describe('ToDoItem component', () => {
  it('snapshot should be matched', () => {
    const { container } = render(
      <Provider store={store}>
        <ToDoItem {...props} />
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('checkbox, text, delete button should be dispayed', () => {
    render(
      <Provider store={store}>
        <ToDoItem {...props} />
      </Provider>,
    );

    expect(screen.getByRole('todo-item-delete-button')).toBeInTheDocument();
    expect(screen.getByRole('todo-item-text')).toBeInTheDocument();
    expect(screen.getByRole('todo-item-checkbox')).toBeInTheDocument();
  });
});
