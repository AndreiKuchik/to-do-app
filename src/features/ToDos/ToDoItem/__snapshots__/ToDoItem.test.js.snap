// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoItem component snapshot should be matched 1`] = `
<div>
  <input
    class="toDoCheckbox"
    role="todo-item-checkbox"
    type="checkbox"
  />
  <div
    class="toDoText false"
    role="todo-item-text"
  >
    ToDo1
  </div>
  <button
    class="deleteButton"
    role="todo-item-delete-button"
    type="button"
  >
    Delete
  </button>
</div>
`;
